asterius/src/Asterius/FrontendPlugin.hs

```diff
@@ -31,14 +32,25 @@ frontendPlugin = makeFrontendPlugin $ do
     void
       $ GHC.setSessionDynFlags
           dflags
-            { GHC.hooks = (GHC.hooks dflags)
-                { GHC.dsForeignsHook = Just asteriusDsForeigns,
-                  GHC.tcForeignImportsHook = Just
-                                               asteriusTcForeignImports,
-                  GHC.tcForeignExportsHook = Just
-                                               asteriusTcForeignExports
-                  }
+            { GHC.hooks =
+                (GHC.hooks dflags)
+                  { GHC.dsForeignsHook = Just asteriusDsForeigns,
+                    GHC.tcForeignImportsHook = Just asteriusTcForeignImports,
+                    GHC.tcForeignExportsHook = Just asteriusTcForeignExports,
+                    GHC.hscCompileCoreExprHook = Just asteriusHscCompileCoreExpr,
+                    GHC.startIServHook = Just asteriusStartIServ,
+                    GHC.iservCallHook = Just asteriusIservCall,
+                    GHC.readIServHook = Just asteriusReadIServ,
+                    GHC.writeIServHook = Just asteriusWriteIServ,
+                    GHC.stopIServHook = Just asteriusStopIServ
+                    }
               }
+  do
+    dflags <- GHC.getSessionDynFlags
+    void
+      $ GHC.setSessionDynFlags
+      $ dflags {GHC.settings = (GHC.settings dflags) {GHC.sPgm_i = "false"}}
```

Comment:
what is the `sPgm_i` option and why do we need to set it?

Reply:
`sPgm_i` is the path to the `iserv` executable. Since we use hooks to override all iserv logic, we don't need a standard iserv process, so we set it to `false` here; just in case ghc start `false` by accident, it'll exit by non-zero immediately to crash ghc, so we can spot where's iserv-related code which we forgot to override.